src/find-changes.ts

```diff
@@ -34,44 +34,76 @@ async function execCommand(
 }
 
 export async function getBranchPoint(context: Context): Promise<string> {
-  if (process.env['GITHUB_EVENT_NAME'] === 'pull_request') {
-    const eventPath = process.env['GITHUB_EVENT_PATH']
-    core.info(`Reading event from ${eventPath}`)
-    if (!eventPath) {
-      throw new Error(
-        'Could not find event payload file to determine branch point.'
-      )
-    }
-    const eventData: Buffer = await fs.readFile(eventPath)
-    const event = JSON.parse(eventData.toString())
-    if (event && event.pull_request) {
-      if (
-        !context.fromOriginalBranchPoint &&
-        event &&
-        event.repository &&
-        event.repository.default_branch
-      ) {
+  const eventName = process.env['GITHUB_EVENT_NAME']
+
+  switch (eventName as string) {
+    case 'pull_request':
+      return handlePullRequest(context)
+    case 'push':
+      return handlePush()
+  }
+  throw new Error(
+    'find-changed-packages only works on pull_request and push events'
+  )
+}
+
+async function handlePullRequest(context: Context): Promise<string> {
+  const event = await getEvent()
+  if (event.action === 'closed') {
+    throw new Error(
+      'Running find-changes on: pull_request: closed is not supported in v2 - please migrate workflow to on: push:'
+    )
+  }
+  const BRANCH = true
+  const MASTER = false
+  switch (context.fromOriginalBranchPoint) {
+    case MASTER:
+      if (event.repository && event.repository.default_branch) {
         const upstream = `origin/${event.repository.default_branch}`
         core.info(`Found branch point ${upstream}`)
         return upstream
-      } else if (
-        context.fromOriginalBranchPoint &&
-        event &&
+      }
+      break
+    case BRANCH:
+      if (
         event.pull_request &&
         event.pull_request.base &&
         event.pull_request.base.sha
       ) {
         core.info(`Found branch point ${event.pull_request.base.sha}`)
         return event.pull_request.base.sha as string
-      } else {
-        throw new Error('Unable to determine branch point to compare changes.')
       }
-    } else {
-      throw new Error('Event payload does not provide the pull request data.')
-    }
-  } else {
-    throw new Error('find-changed-packages only works on pull_request events')
+      break
+  }
+  throw new Error(
+    'Unable to determine pull request branch point to compare changes.'
+  )
+}
+
+async function handlePush(): Promise<string> {
+  const event = await getEvent()
+  if (event.before) {
+    core.info(`Found branch point ${event.before}`)
```

Comment:
In all merge scenarios (merge commit, rebase and squash), will the diff between `event.before` and the latest (current CI/release run) be *only* the changes from the merged PR

Reply:
To my understanding, yes. I'll do some additional manual tests (always had the same filename being added or removed) but in my other playing around with the before it behaved as expected.

One concern I had is that merge commit may have changes from another PR, but I guess the merge commit might change some of those files so it's desired behavior.

Reply:
> that merge commit may have changes from another PR

This should only be possible if the merge commit was a conflict resolution, which *shouldn't* be the case in PR merges. In normal use with merging a PR from github UI, I would expect the changes in the merge commit to be empty

Reply:
I think in effect we are interested in `git diff HEAD^..HEAD` after the merge, whereas what we were doing before my bugfix for multiple merges is `git diff HEAD^...HEAD` which would diff from the original branch point, rather than the commit immediately prior to merging.

As long as we are comparing the merge to the immediately prior commit on the main branch, I think we're probably good.

Reply:
discussed in slack, resolving